app/src/sharedTest/java/org/oppia/app/activity/InputInteractionViewTestActivityTest.kt

```diff
@@ -0,0 +1,88 @@
+package org.oppia.app.activity
+
+import android.app.Activity
+import android.content.Intent
+import android.content.pm.ActivityInfo
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.intent.Intents
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.rule.ActivityTestRule
+import org.junit.Assert.*
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.app.R
+import org.oppia.app.model.InteractionObject
+import androidx.test.core.app.ActivityScenario
+import androidx.test.espresso.action.ViewActions
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import org.hamcrest.CoreMatchers.`is`
+import org.hamcrest.Matchers.instanceOf
+import org.junit.After
+import org.oppia.app.customview.inputInteractionView.NumericInputInteractionView
+
+/** Tests for [InputInteractionViewTestActivity]. */
+@RunWith(AndroidJUnit4::class)
+class InputInteractionViewTestActivityTest {
+  private lateinit var launchedActivity: Activity
+  @get:Rule
+  var activityTestRule: ActivityTestRule<InputInteractionViewTestActivity> = ActivityTestRule(
+    InputInteractionViewTestActivity::class.java, /* initialTouchMode= */ true, /* launchActivity= */ false
+  )
+
+  @Test
+  fun testNumericInputInteractionView_withNoInputText_hasCorrectPendingAnswerType() {
+    val activityScenario = ActivityScenario.launch(InputInteractionViewTestActivity::class.java)
+    activityScenario.onActivity { activity ->
+      val textAnswerRetriever =
+        activity.findViewById(R.id.test_number_input_interaction_view) as NumericInputInteractionView
+      assertThat(textAnswerRetriever.getPendingAnswer(), instanceOf(InteractionObject::class.java))
+    }
+  }
+
+  @Test
+  fun testNumericInputInteractionView_withInputtedText_hasCorrectPendingAnswer() {
+    val activityScenario = ActivityScenario.launch(InputInteractionViewTestActivity::class.java)
+    onView(withId(R.id.test_number_input_interaction_view)).perform(ViewActions.clearText(), ViewActions.typeText("9"))
+    activityScenario.onActivity { activity ->
+      val textAnswerRetriever =
+        activity.findViewById(R.id.test_number_input_interaction_view) as NumericInputInteractionView
+      assertEquals(textAnswerRetriever.getPendingAnswer().objectTypeCase, InteractionObject.ObjectTypeCase.REAL)
+      assertThat(textAnswerRetriever.getPendingAnswer().real, `is`("9".toDouble()))
+    }
+  }
+
+  @Test
+  fun testNumericInputInteractionView_withInputtedText_hasCorrectPendingAnswerWithDecimalValues() {
+    val activityScenario = ActivityScenario.launch(InputInteractionViewTestActivity::class.java)
+    onView(withId(R.id.test_number_input_interaction_view)).perform(
+      ViewActions.clearText(),
+      ViewActions.typeText("9.5")
```

Comment:
Nit here & elsewhere: prefer statically importing typeText() to be more inline with the fluent patterns of Espresso (i.e. it helps the flow of onView()/withId(), etc. read better).

Reply:
Resolved